client/components/theme-collection/index.tsx

```diff
@@ -0,0 +1,110 @@
+
+	const [ isSwiperLoaded, setSwiperLoaded ] = useState( false );
+	useEffect( () => {
+		if ( ! isSwiperLoaded ) {
+			const el = document.querySelector( `#${ swiperContainerId }` ) as HTMLElement;
+			if ( el ) {
+				/**
+				 * We have to import the swiper package dynamically because it doesn't offer a CommonJS version. Because of this, the SSR build will fail.
+				 *
+				 * From a performance standpoint, it should not cause any issues; we are already loading the ThemeCards dynamically anyway.
+				 */
+				Promise.all( [
+					import( 'swiper' ),
+					import( 'swiper/modules' ),
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					import( 'swiper/css' ),
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					import( 'swiper/css/navigation' ),
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					import( 'swiper/css/keyboard' ),
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore
+					import( 'swiper/css/mousewheel' ),
+				] ).then( ( values ) => {
+					const Swiper = values[ 0 ].default;
+					const { Navigation, Keyboard, Mousewheel } = values[ 1 ];
+
+					swiperInstance.current = new Swiper( el, {
+						cssMode: true,
+						mousewheel: true,
+						keyboard: {
+							enabled: true,
+							onlyInViewport: false,
+						},
+						navigation: {
+							nextEl: '.theme-collection__carousel-nav-button--next',
+							prevEl: '.theme-collection__carousel-nav-button--previous',
+						},
+						threshold: 5,
+						slideToClickedSlide: true,
+						slidesPerView: 'auto',
+						spaceBetween: 20,
+						modules: [ Navigation, Keyboard, Mousewheel ],
+					} );
+					setSwiperLoaded( true );
+				} );
+			}
+		}
+
+		return () => {
+			swiperInstance.current?.destroy();
+		};
+	}, [ swiperContainerId, isSwiperLoaded ] );
+
+	return (
+		<div className="theme-collection__container ">
+			<h2>{ heading }</h2>
+			{ subheading }
+			<div className="swiper-container" id={ swiperContainerId }>
+				<div className="theme-collection__carousel-controls">
+					<Button className="theme-collection__carousel-nav-button theme-collection__carousel-nav-button--previous">
+						<Icon icon={ chevronLeft } />
+					</Button>
+					<Button className="theme-collection__carousel-nav-button theme-collection__carousel-nav-button--next">
+						<Icon icon={ chevronRight } />
+					</Button>
+				</div>
+				<div className="theme-collection__list-wrapper swiper-wrapper">{ children }</div>
+			</div>
+		</div>
+	);
+}
+
+export function CollectionListItem( {
+	collectionSlug,
+	themeId,
+	children,
+}: PropsWithChildren< { collectionSlug: string; themeId: string } > ) {
+	return (
+		<div
+			key={ `theme-collection-container-${ collectionSlug }-${ themeId }` }
+			className="theme--collection__list-item swiper-slide"
+		>
+			{ children }
+		</div>
+	);
+}
```

Comment:
What's the reasoning for `CollectionListItem`?

For what I can see, the only significant bit is the `swiper-slide` CSS class needed by Swiper.
Everything else seems superfluous, and even `swiper-slide` technically is. 😄 

If you need to style a slide (e.g. to give them a width), you can do that with `.theme-collection__list-wrapper > *`.
It then becomes the responsibility of the consumer to ensure the ThemeCollection children have an appropriate/stylable wrapper div.

If you need a specific slide class for the Swiper initialization, why not let the consumer define it?

e.g.
```jsx
// consumer
<ThemeCollection slideClass="foobar__slide" />

// Collection
function ThemeCollection( { slideClass } ) {
  // ...
  new Swiper( el, {
    // all the settings
    slideClass
  } );
  // ...
}
```

This would also give us the opportunity to offer further slider customizations to various Collection consumers (not the case right now, but it's good to keep in mind for the future).

Reply:
As a counterpoint, I think there's value in offering a `ThemeCollectionItem` component, which might decrease the consumer's cognitive load.

I'd recommend keeping it as simple as possible, with only the essential bits for now and maybe extending it in the future if needed. E.g.

```jsx
export default function ThemeCollectionItem( { children } ) {
  return (
    <div className="theme-collection__item swiper-slide">
      { children }
    </div>
  );
}
```